tests/loggers/test_remote_uploader_downloader.py

```diff
@@ -119,15 +119,21 @@ def object_store_test_helper(
                     # Give it some time to finish the second upload
                     # (Since the upload is really a file copy, it should be fast)
                     time.sleep(2)
-                    # Then, crashes are detected on the next batch end / epoch end event
-                    with pytest.raises(RuntimeError):
+                    # Then, crashes are detected on the next batch end, but
+                    # should be a FileExistsError not a runtime error because the parent will raise
+                    # the fatal exception that the worker throws.
+                    with pytest.raises(
+                            FileExistsError,
+                            match=
+                            f'Object local://{remote_file_name} already exists, but allow_overwrite was set to False.'):
                         remote_uploader_downloader.run_event(Event.BATCH_END, dummy_state, logger)
 
-                    with pytest.raises(RuntimeError):
```

Comment:
this is no longer an error?

Reply:
it is, but the BATCH_END call would normally exit the process. It's only a technicality that pytest catches it and technically we could run another event for EPOCH_END. I updated it to separately test BATCH_END and EPOCH_END with a parametrization